homework/week-0/AllisonBojarski/AnimalSort(HWproblem#1)/src/nyc/c4q/alizinha/DomesticCat.java

```diff
@@ -0,0 +1,37 @@
+package nyc.c4q.alizinha;
+/**
+ * Created by c4q-Allison on 5/1/15.
+ */
+public class DomesticCat extends Animal implements Domestic
+{
+
+    String name;
+
+    @Override
+    public String getName()
+    {
+        return this.name;
+    }
+
+    @Override
+    public void setName(String name)
+    {
+        this.name = name;
+    }
+
+    @Override
+    public int compareTo(Animal o)
+    {
+        if (o instanceof Domestic) {
```

Comment:
I'm a bit confused by this. Your Animal reference o is of type Domestic? Isn't Domestic an interface? Would this always return true? If so, would it be necessary here?

I think that if you create a reference of type DomesticCat and set it equal to (DomesticCat) animal, you could use this.getName to compare to animal.getName. An example of this:

DomesticCat mrBigglesworth = (DomesticCat) animal;